src/vm/gcenv.ee.cpp

```diff
@@ -1169,3 +1169,13 @@ void GCToEEInterface::FreeStringConfigValue(const char* value)
 {
     delete [] value;
 }
+
+bool GCToEEInterface::IsGCThread()
+{
+    return ::IsGCThread();
```

Comment:
You also need to fix `ClrFlsSetThreadType` to make this work reliably.

Reply:
Ok, I see you have it in the PR description that you will take care of it later.

Reply:
Yep! The code for that is in `gc_heap::gc_thread_stub` and it should be calling into the EE to set this up instead of doing it in that function.

One wrinkle is that server GC threads are created by `GCToOSInterface::CreateThread` while background GC threads are created by `GCToEEInterface::CreateBackgroundThread`. It seems to me that these could be combined into a single `GCToEEInterface::CreateThread` that creates a thread, does `ClrFlsSetThreadType`, and calls the provided GC thread stubs. The only difference that I can tell between background GC threads and server GC threads is that the EE background thread GC stub sets the GCSpecial bit on its `Thread` object.

Otherwise, I suppose we could create a new method like `GCToEEInterface::RegisterThread` that informs the EE of a new GC thread created through `GCToOSInterface::CreateThread` (ultimately doing `ClrFlsSetThreadType(ThreadType_GC)`.

@Maoni0 do you have any thoughts on this?